### PR TITLE
Allow pytest disable warnings in CI

### PR DESCRIPTION
**Pull request by dstandish**

### CODE REVIEW

File: dev/breeze/src/airflow_breeze/commands/main_command.py [hunk 1]
```diff
@@ -155,6 +155,8 @@ def check_for_python_emulation():
 def check_for_rosetta_environment():
     if sys.platform != "darwin":
         return
+    if platform.processor() == "i386":
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/commands/main_command.py]: That's likely a. left-over?

File: dev/breeze/src/airflow_breeze/commands/testing_commands.py [hunk 2]
```diff
@@ -72,6 +72,8 @@
 
 LOW_MEMORY_CONDITION = 8 * 1024 * 1024 * 1024
 
+DISABLE_WARNINGS_LABEL = "disable warnings"
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/commands/testing_commands.py]: I believ we wanted to "enable warnings" by label so that they are hidden by default ?

Comment (dstandish) [reply dev/breeze/src/airflow_breeze/commands/testing_commands.py]: even better --- i thought you wanted them there by default

File: dev/breeze/src/airflow_breeze/commands/main_command.py (hunk 1 above)
Comment (dstandish) [reply dev/breeze/src/airflow_breeze/commands/main_command.py]: it blew up in this check for me.  i think to repro, you would do ... use intel mac, change a command.  
i think we need something like this

Comment (dstandish) [reply dev/breeze/src/airflow_breeze/commands/main_command.py]: here is traceback
```
Default value of mssql_version parameter 2017-latest used.

Traceback (most recent call last):
  File "/Users/dstandish/code/airflow/dev/breeze/src/airflow_breeze/commands/main_command.py", line 159, in check_for_rosetta_environment
    runs_in_rosetta = subprocess.check_output(
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['sysctl', '-n', 'sysctl.proc_translated']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/dstandish/code/airflow/dev/breeze/src/airflow_breeze/breeze.py", line 49, in <module>
    main()
  File "/Users/dstandish/.cache/pre-commit/repo9d4d9t30/py_env-python3/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/dstandish/.cache/pre-commit/repo9d4d9t30/py_env-python3/lib/python3.8/site-packages/rich_click/rich_group.py", line 21, in main
    rv = super().main(*args, standalone_mode=False, **kwargs)
  File "/Users/dstandish/.cache/pre-commit/repo9d4d9t30/py_env-python3/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/dstandish/.cache/pre-commit/repo9d4d9t30/py_env-python3/lib/python3.8/site-packages/click/core.py", line 1654, in invoke
    super().invoke(ctx)
  File "/Users/dstandish/.cache/pre-commit/repo9d4d9t30/py_env-python3/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/dstandish/.cache/pre-commit/repo9d4d9t30/py_env-python3/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/dstandish/.cache/pre-commit/repo9d4d9t30/py_env-python3/lib/python3.8/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/dstandish/code/airflow/dev/breeze/src/airflow_breeze/commands/main_command.py", line 113, in main
    check_for_rosetta_environment()
  File "/Users/dstandish/code/airflow/dev/breeze/src/airflow_breeze/commands/main_command.py", line 192, in check_for_rosetta_environment
    except TimeoutOccurred:
UnboundLocalError: local variable 'TimeoutOccurred' referenced before assignment
```

**Review by potiuk — COMMENTED** [review]
One nit, one doubt. 

I like it this way - i wanted to do it via selective-checks, but this way is indeed simpler and easy to add other test variations this way.

File: dev/breeze/src/airflow_breeze/commands/main_command.py (hunk 1 above)
Comment (potiuk) [reply dev/breeze/src/airflow_breeze/commands/main_command.py]: Ah. Right. Should we separate it to another PR - it's not really related :) ?

File: dev/breeze/src/airflow_breeze/commands/testing_commands.py (hunk 2 above)
Comment (potiuk) [reply dev/breeze/src/airflow_breeze/commands/testing_commands.py]: Not really. I think it was more of @uranusjr's quest to fix them and I think he turned them on in the first place, for that reason, but I think indeed this should be more ore less dedicated effort to remove them and showing them by default when we have many of them do not play well with GitHub UI.

**Review by potiuk — APPROVED** [review]
With nit for splitting.

File: dev/breeze/src/airflow_breeze/commands/testing_commands.py (hunk 2 above)
Comment (dstandish) [reply dev/breeze/src/airflow_breeze/commands/testing_commands.py]: yeah, i agree.  i also appreciate the desire to squash those warnings. there's a lot of noise... and if it shows up in the tests, probably a lot of it shows up for users.

**Review by ashb — COMMENTED** [review]
Let's not? Let's remove/fix the warnings instead?

**Comment by potiuk** [comment]
@ash - Absolutely. I'd'love that.

What do you propose exactly ? How do you think we can improve curent situation?

Currently the warnings are printed and besides slowing the ways how people can se their test failures, they don't do much - that's the nature of warnings. It's lile that for quite some time.

I have my own idea but it's quite disruptive and might be difficult to accept so I would love to hear, maybe others have better ideas before I bias them with mine.

**Comment by ashb** [comment]
My idea is a Two phase approach, similar to how we did... Typing? in the past. And I call it a ratchet.

The first step is to capture how many warnings each test file produces, and then we fail ci if any new warnings are issued.

The next step is once a file doesn't have any warnings there is a way of making pytest treat warnings as failure (I think we do it in one place in our test site right now).

I noticed the sheer number of warnings when looking at the bug fix for rc3: it's way up from where I remember it, and it was only my list to look at next week. I suspect over half are from one of two helper funcs in our test suite!

**Comment by potiuk** [comment]
Yes. This is good idea :).

**Comment by potiuk** [comment]
Closing as we have better solution in #28096 . But .. follow up on actually fixing the warnings is a good idea :)

**Pull request closed without merging**